pkg/agent/agent.go

```diff
@@ -663,6 +667,16 @@ func (i *Initializer) configureGatewayInterface(gatewayIface *interfacestore.Int
 		return err
 	}
 
+	if i.disableTXChecksumOffload {
+		if err := ethtool.EthtoolTXHWCsumSwitch(i.hostGateway, ethtool.TXCSUM_OFF); err != nil {
+			return fmt.Errorf("error when disabling TX checksum offload on %s: %v", i.hostGateway, err)
+		}
+	} else {
+		if err := ethtool.EthtoolTXHWCsumSwitch(i.hostGateway, ethtool.TXCSUM_ON); err != nil {
```

Comment:
Can you check if tx checksum is always turned on by default on linux platforms and always configurable?
If not, this change would affect cases in which tx checksum was off, but the default value "disabledTXChecksumOffload=false" would try to turn it on.

Reply:
I have investigated about `TX Checksum Offload` and found that `TX Checksum Offload` is not always supported since the driver of a NIC may not supports that. I thank `TX Checksum Offload` is turned on default on Linux if the driver of a NIC supports `TX Checksum Offload`. 

Maybe we can remove this since `TX Checksum Offload` cannot be set to on in some environment. However, I don't have a good idea how to deal with the case:

A user sets `disabledTXChecksumOffload=true`, then sets it back to false, but `TX Checksum Offload` of Antrea gateway is still off, maybe we could log something like "If you set disabledTXChecksumOffload=false back again and you should set TX Checksum Offload of Antrea gateway to on manually"?